### PR TITLE
Feature: LogViewer pagination

### PR DESCRIPTION
**Pull request by loivsen**
## Description

Pagination added for common log messages (10 per page) and saved searches (15 per page). Can be changed

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

### CODE REVIEW

File: src/packages/log-viewer/workspace/views/overview/components/log-viewer-message-templates-overview.element.ts [hunk 1]
```diff
@@ -1,39 +1,52 @@
...
-			this._messageTemplates = templates ?? null;
+			this._messageTemplates = templates?.items ?? [];
+			this._total = templates?.total ?? 0;
 		});
 	}
 
 	#getMessageTemplates() {
-		const take = this._messageTemplates?.items?.length ?? 0;
-		this.#logViewerContext?.getMessageTemplates(0, take + 10);
+		//const take = this._messageTemplates?.length ?? 0;
```
Comment (iOvergaard) [inline src/packages/log-viewer/workspace/views/overview/components/log-viewer-message-templates-overview.element.ts:42]: I think it's safe to remove this line :)

```suggestion
```

File: src/packages/log-viewer/workspace/views/overview/components/log-viewer-message-templates-overview.element.ts [hunk 2]
```diff
@@ -54,11 +67,11 @@ export class UmbLogViewerMessageTemplatesOverviewElement extends UmbLitElement {
 	render() {
 		return html`
 			<uui-box headline="Common Log Messages" id="saved-searches">
-				<p style="font-style: italic;">Total Unique Message types: ${this._messageTemplates?.total}</p>
+				<p style="font-style: italic;">Total Unique Message types: ${this._total}</p>
```
Comment (iOvergaard) [inline src/packages/log-viewer/workspace/views/overview/components/log-viewer-message-templates-overview.element.ts:70]: Do we have a localization key for this or is it new?

**Review by iOvergaard — CHANGES_REQUESTED** [review]
Looking good - just found a few things. Having tested the saved searches, I don't believe we can make a pagination for it at the moment since it uses a shared context across the whole logviewer workspace. We are showing the saved searches in a dropdown in the Search view, which messes with the pagination in the Overview view:

image

Honestly, I think you should set the maximum of saved searches to something like 999 and then remove the uui-pagination. It's one of those edge cases where we'd just like to show everything if possible.

**Comment by iOvergaard** [comment]
Also -- are you planning to build pagination for the "Search" view as well? I think that would be the biggest/best addition to the log viewer workspace.

**Comment by loivsen** [comment]
> Also -- are you planning to build pagination for the "Search" view as well? I think that would be the biggest/best addition to the log viewer workspace.

Doesn't it already have pagination? or am i looking at the wrong place?
image

File: src/packages/log-viewer/workspace/views/overview/components/log-viewer-message-templates-overview.element.ts (hunk 2 above)
Comment (loivsen) [reply src/packages/log-viewer/workspace/views/overview/components/log-viewer-message-templates-overview.element.ts:69]: This message exists in old Backoffice, but it doesn't seem like there is any localization keys for it. There seems to be more things missing for log viewer like "common log messages" and "number of errors", etc.

**Comment by iOvergaard** [comment]
> > Also -- are you planning to build pagination for the "Search" view as well? I think that would be the biggest/best addition to the log viewer workspace.
> 
> Doesn't it already have pagination? or am i looking at the wrong place?

Cool, I must have tested with not enough data. Carry on 🥇

**Comment by loivsen** [comment]
> Looking good - just found a few things. Having tested the saved searches, I don't believe we can make a pagination for it at the moment since it uses a shared context across the whole logviewer workspace. We are showing the saved searches in a dropdown in the Search view, which messes with the pagination in the Overview view:
> 
> image
> Honestly, I think you should set the maximum of saved searches to something like 999 and then remove the uui-pagination. It's one of those edge cases where we'd just like to show everything if possible.

Alternatively we could also just add pagination to the other places like the dropdown in the search.
I changed `take` to 999 for now and letting the pagination logic stay in case we change our mind later down the line. It doesn't render pagination anyway unless the limit is reached (999 now in this case).

**Comment by iOvergaard** [comment]
> > Looking good - just found a few things. Having tested the saved searches, I don't believe we can make a pagination for it at the moment since it uses a shared context across the whole logviewer workspace. We are showing the saved searches in a dropdown in the Search view, which messes with the pagination in the Overview view:
> Alternatively we could also just add pagination to the other places like the dropdown in the search. I changed `take` to 999 for now and letting the pagination logic stay in case we change our mind later down the line. It doesn't render pagination anyway unless the limit is reached (999 now in this case).

Yeah, but I guess we need to split the context a bit, since currently, we have one Observable for `savedSearches` in the context, so that means if you change to page 2 in one place, it will show the page 2 content in the other place, but the pagination won't update automatically.

Good idea to fix it for now with what you just committed 👍

**Pull request merged**